### PR TITLE
add responsive web design for small screens

### PR DESCRIPTION
**Pull request by evoludolab**
This PR is an attempt at improving support for small screens in general and phones in particular. This adds responsive web design features to adjust the size of albums and photo thumbnails based on size of screen. It also adjusts margins and title size as well as suppresses subtitles to improve visibility of small thumbnails.

As an aside, I have some further ideas for improving the user experience on touch devices (for example, toggling the overlay on images doesn't seem to work for touch). Let me know if changes in this direction are of interest.

### CODE REVIEW

**Comment by evoludolab** [comment]
These modest beginnings are starting to get exciting 😃 but they are also turning into a bigger project than anticipated... Currently reworking the about panel for small screens. This will require a couple more commits before the PR is ready. I will report back once the changes are ready for review. In the meantime any thoughts/comments/suggestions are most welcome.

**Comment by ildyria** [comment]
Hi @evoludolab sorry for being slow to answer pretty busy IRL at the moment. That being said we really do appreciate your efforts. :+1:

**Review by ildyria — COMMENTED** [review]
Not tested yet, but the code looks nice. :+1:

**Comment by evoludolab** [comment]
@ildyria No worries and thanks for the encouraging feedback. Now I am finally done with this PR! Note that some small changes in the back end are required as well, see LycheeOrg/Lychee#939.

All changes are almost exclusively restricted to small devices. Only two changes affect the general appearance in very small ways: first, a new mode for the header/toolbar is added for all settings pages (omits all the buttons at the top right that don't make sense in this context) and second, the 'x' to cancel search is properly placed.

In contrast, much has changed for small screens:

1. the info panel changes depending on size and orientation of the device 

    1.  on very small screens the info panel is a semi-transparent overlay
    2. on medium sized screens the info panel moves to the side as usual but uses smaller fonts etc.
    3. no change on normal screens

2. the settings menu changes for small(er) devices: instead of 'wasting' screen space with the left menu it opens a context menu. This requires a new header/toolbar with a close button to return to the album.

3. the layout consistently adapts to small devices.

In parallel I have been preparing a PR that improves the interface for touch devices (I'll update with a link once I prepared the PR). Unfortunately the two PRs most likely result in (easy) merge conflicts. If it helps, I have them merged on my machine and could lump them together into a single PR.

**Comment by evoludolab** [comment]
As an aside, this PR at least partially addresses #66 as well as #162. In particular, the layout adapts well to small screens for *justified* and *square* layouts (while it's not immediately obvious how *unjustified* should behave).

**Comment by ildyria** [comment]
@evoludolab Sorry for the delay, is this ready for review ? :)

**Comment by evoludolab** [comment]
Yeah, this one is ready, too. Looking forward to your review.

File: scripts/main/view.js [hunk 1]
```diff
@@ -580,7 +580,7 @@ view.album = {
 	},
 
 	sidebar: function () {
-		if ((visible.album() || !album.json.init) && !visible.photo()) {
+		if ((visible.album() || (album.json !== null && album.json.init !== null)) && !visible.photo()) {
```
Comment (ildyria) [inline scripts/main/view.js:583]: can't this be simplified in:
```suggestion
		if ((visible.album() || (album.json && album.json.init)) && !visible.photo()) {
```

**Review by ildyria — APPROVED** [review]
LGTM. Tested.

File: scripts/main/view.js (hunk 1 above)
Comment (ildyria) [reply scripts/main/view.js:583]: @evoludolab can you check that one?
Otherwise, this is a really good job. Ready to merge IMHO.

**Comment by evoludolab** [comment]
@ildyria Thanks for the feedback and the suggestion. I have applied it. The reason was likely that I usually code in java, which requires explicit checks for null...

**Comment by evoludolab** [comment]
@ildyria Thanks. Since #254 now has conflicts I suggest that I first sort them out and send you a heads-up when done.

**Pull request merged**